README.md

```diff
@@ -52,55 +55,98 @@ accumulated remain preserved and accessible over time.
+data to `Object[][]` type in the `UpdateViewUseCaseInteractor` to accommodate Java Swing requirements. 
+However, we thought it is not adhering to SOLID principle from the point that `Object[][]` may not be the type required 
+by other view libraries that could be imported, which is in contrary to the Open/Closed Principle.
+- `SessionStorage` is an object type used to hold all information contained in the user's session. Every time a use case 
+is called, the interactor of that use case needs to access the information within the `SessionStorage` object. Thus, it 
 needs to be passed throughout all layers of our program, despite being a violation of Clean Architecture.
 
 ### SOLID Principles
 - **Single Responsibility Principle:** All classes are acted upon by a single actor and adhere to this principle.
-- **Open-Closed Principle:** Since our classes with main functionality implement IB and OB interfaces, the user is able 
-to modify any interactor or presenter as they see fit. For example, if the user wanted to implement another view that 
-generates statistical data with all months they could create another interactor to do that.
-- **Liskov Substitution Principle:** All of our IB and OB objects adhere to this principle. In order to adhere to the 
-DIP, the interactors and presenters are cast to the IB and OB, respectively. This demonstrates the LSP, as any class 
-that implements these interfaces can be used. Another example is the LoadMonthMenuVB, which 
+- **Open-Closed Principle:** Since our classes with main functionality implement Input Boundary and Output Boundary
+interfaces, the user is able to modify any interactor or presenter as they see fit. For example, if the user wanted to 
+implement another view that generates statistical data with all months they could create another interactor to do that.
+- **Liskov Substitution Principle:** All of our Input Boundary and Output Boundary objects adhere to this principle. 
+In order to adhere to the Dependency Inversion Principle, the interactors and presenters are cast to the Input Boundary 
+and Output Boundary, respectively. This demonstrates the LSP, as any class that implements these interfaces can be used.
 - **Interface Segregation Principle:** All of our interfaces adhere to this rule, as they are all small and contain only
 the necessary methods needed.
 - **Dependency Inversion Principle:** All classes have a boundary between every layer in Clean Architecture, to create 
 the dependency inversion.
 
 ### Design Patterns
-- The Facade design pattern is used to implement the GenerateSummaryUCI to delegate tasks to the 
-GenerateSummaryUCInterpreter for generating the statistical data, and the GenerateSummaryP for returning it.
-- In the future, we could implement the Factory design pattern to implement the Add/Edit Categories and Add/Edit 
+- The Factory design pattern is used to implement the Add/Edit Categories and Add/Edit 
```

Comment:
Also I think we should write more for these, because this is where we are supposed to be explaining our choices! Something like:

"The Factory design pattern is used to implement the Add/Edit Category and Add/Edit Expense features. We chose to add this implementation to avoid tight coupling between the creators (ExpenseUseCaseInteractor and CategoryUseCaseInteractor) and their products (Expense and Category). This choice helps our code adhere to the Open/Closed Principle, since we can introduce new types of MonthObjects into the code and avoid breaking our current implementation. Additionally, it adheres to the Single Responsibility Principle, as the product creation code is moved to a separate class, allowing the UseCaseInteractors to focus on error handling and post-creation tasks."

Feel free to add or modify!

Reply:
Ohh, thank you for pointing this out. I integrated this paragraph now!